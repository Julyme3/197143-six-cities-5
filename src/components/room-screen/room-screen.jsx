import React from "react";
import {connect} from "react-redux";
import {OfferType, ReviewsType, OffersType} from "../../types";
import withForm from "../../hocs/with-form/with-form";
import ReviewForm from "../review-form/review-form";
import ReviewsList from "../reviews-list/reviews-list";
import Map from "../map/map";
import OfferListNear from "../offer-list-near/offer-list-near";
import {getNearestOffers} from "../../offers";
import {Type} from "../../const";
import MainLayout from "../../layouts/main-layout/main-layout";

const ReviewFormWrapped = withForm(ReviewForm);
const RoomScreen = (props) => {
  const {offer, reviews} = props;
  const {details, user} = offer;
  const nearestOffers = getNearestOffers(props.offers.filter((item) => item.typeComponent === Type.NEAR), offer);

  return (
    <MainLayout
      classNameWrap="page__main--property"
    >
      <section className="property">
        <div className="property__gallery-container container">
          <div className="property__gallery">
            {offer.src.map((srcItem, i) => {
              return (
                <div
                  key={`${i}-${srcItem}`}
                  className="property__image-wrapper"
                >
                  <img className="property__image" src={srcItem} alt="Photo studio" />
                </div>
              );
            })}
          </div>
        </div>
        <div className="property__container container">
          <div className="property__wrapper">
            {offer.isPremium &&
              <div className="property__mark">
                <span>Premium</span>
              </div>
            }
            <div className="property__name-wrapper">
              <h1 className="property__name">
                {offer.name}
              </h1>
              <button className="property__bookmark-button button" type="button">
                <svg className="property__bookmark-icon" width="31" height="33">
                  <use xlinkHref="#icon-bookmark"></use>
                </svg>
                <span className="visually-hidden">To bookmarks</span>
              </button>
            </div>
            <div className="property__rating rating">
              <div className="property__stars rating__stars">
                <span style={{width: `80%`}}></span>
                <span className="visually-hidden">Rating</span>
              </div>
              <span className="property__rating-value rating__value">{offer.raiting}</span>
            </div>
            <ul className="property__features">
              <li className="property__feature property__feature--entire">
                {offer.type}
              </li>
              <li className="property__feature property__feature--bedrooms">
                {details.bedrooms} Bedrooms
              </li>
              <li className="property__feature property__feature--adults">
                Max {details.adults} adults
              </li>
            </ul>
            <div className="property__price">
              <b className="property__price-value">€{offer.price}</b>
              <span className="property__price-text">&nbsp;night</span>
            </div>
            <div className="property__inside">
              <h2 className="property__inside-title"></h2>
              <ul className="property__inside-list">
                {details.insideItems.map((insideItem, i) => {
                  return (
                    <li
                      key={`${i}-${insideItem}`}
                      className="property__inside-item">
                      {insideItem}
                    </li>
                  );
                })}
              </ul>
            </div>
            <div className="property__host">
              <h2 className="property__host-title">Meet the host</h2>
              <div className="property__host-user user">
                <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                  <img className="property__avatar user__avatar" src={user.src} width="74" height="74" alt="Host avatar" />
                </div>
                <span className="property__user-name">
                  {user.name}
                </span>
              </div>
              <div className="property__description">
                <p className="property__text">
                  {details.description}
                </p>
              </div>
            </div>
            <section className="property__reviews reviews">
              <h2 className="reviews__title">Reviews · <span className="reviews__amount">{reviews.length}</span></h2>
              <ReviewsList reviews={reviews} />
              <ReviewFormWrapped />
            </section>
          </div>
        </div>
        <section className="property__map map">
          <Map
            offers={[...nearestOffers, offer]}
            width={`100%`}
            height={`579px`}
          />
        </section>
      </section>
      <div className="container">
        <section className="near-places places">
          <h2 className="near-places__title">Other places in the neighbourhood</h2>
          <OfferListNear
            offers={nearestOffers}
          />
        </section>
      </div>
    </MainLayout>
  );
};

RoomScreen.propTypes = {
  offer: OfferType,
  reviews: ReviewsType,
  offers: OffersType,
};

const mapStateToProps = (state) => ({
  offers: state.DATA.offers,
});

export {RoomScreen};

export default connect(mapStateToProps, null)(RoomScreen);
