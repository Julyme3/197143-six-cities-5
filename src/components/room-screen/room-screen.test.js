import React from "react";
import renderer from "react-test-renderer";
import {BrowserRouter} from 'react-router-dom';
import {RoomScreen} from "./room-screen";

jest.mock(`../review-form/review-form`, () => `ReviewForm`);
jest.mock(`../reviews-list/reviews-list`, () => `ReviewsList`);
jest.mock(`../offer-list-near/offer-list-near`, () => `OfferListNear`);
jest.mock(`../../layouts/main-layout/main-layout`, () => `MainLayout`);
jest.mock(`../rating/rating`, () => `Rating`);

const noop = () => {};

const activeOffer = {
  city: {
    location: [50.938361, 6.959974],
    name: `Cologne`,
    zoom: 13,
  },
  previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
  type: `house`,
  price: 493,
  description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  id: 1,
  src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
    `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
  isPremium: false,
  name: `The house among olive `,
  raiting: 3.8,
  isBookmark: false,
  coords: [50.957361, 6.9509739999999995],
  details: {
    bedrooms: 3,
    adults: 4,
    insideItems: [`Laptop friendly workspace`, `Washer`],
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
  },
  user: {
    src: `img/avatar-angelina.jpg`,
    id: 25,
    name: `Angelina`,
    isSuper: true,
  },
  typeComponent: `main`,
};

const reviews = [
  {
    id: 2,
    date: `2020-11-05T13:38:44.753Z`,
    src: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg`,
    name: `Christina`,
    raiting: 4,
    text: `I stayed here for one night and it was an unpleasant experience.`,
    dateFormated: `November 2020`,
  },
  {
    id: 3,
    date: `2020-10-05T13:38:44.753Z`,
    src: `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/12.jpg`,
    name: `Julia`,
    raiting: 2.8,
    text: `The room was spacious and clean.`,
    dateFormated: `October 2020`,
  },
];

const offers = [
  {
    city: {
      location: [50.938361, 6.959974],
      name: `Cologne`,
      zoom: 13,
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    type: `house`,
    price: 493,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    id: 1,
    src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    isPremium: false,
    name: `The house among olive `,
    raiting: 3.8,
    isBookmark: false,
    coords: [50.957361, 6.9509739999999995],
    details: {
      bedrooms: 3,
      adults: 4,
      insideItems: [`Laptop friendly workspace`, `Washer`],
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    },
    user: {
      src: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isSuper: true,
    },
    typeComponent: `near`,
  },
  {
    city: {
      location: [50.938361, 6.959974],
      name: `Cologne`,
      zoom: 13,
    },
    previewImage: `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    type: `house`,
    price: 493,
    description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    id: 2,
    src: [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    isPremium: false,
    name: `The house among olive `,
    raiting: 3.8,
    isBookmark: false,
    coords: [50.957361, 6.9509739999999995],
    details: {
      bedrooms: 3,
      adults: 4,
      insideItems: [`Laptop friendly workspace`, `Washer`],
      description: `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    },
    user: {
      src: `img/avatar-angelina.jpg`,
      id: 25,
      name: `Angelina`,
      isSuper: true,
    },
    typeComponent: `near`,
  },
];

const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

const match = {
  params: {
    id: 2,
  },
};

describe(`RoomScreen render`, () => {
  it(`RoomScreen for no Authorization user`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <RoomScreen
              activeOffer={activeOffer}
              reviews={reviews}
              nearbyOffers={offers}
              fetchFullOfferAction={noop}
              fetchNearbyOffers={noop}
              fetchComments={noop}
              postComment={noop}
              authorizationStatus={AuthorizationStatus.NO_AUTH}
              postFavorite={noop}
              setActiveOffer={noop}
              match={match}
            ><React.Fragment />
            </RoomScreen>
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`RoomScreen for Authorization user`, () => {
    const tree = renderer
      .create(
          <BrowserRouter>
            <RoomScreen
              activeOffer={activeOffer}
              reviews={reviews}
              nearbyOffers={offers}
              fetchFullOfferAction={noop}
              fetchNearbyOffers={noop}
              fetchComments={noop}
              postComment={noop}
              authorizationStatus={AuthorizationStatus.AUTH}
              postFavorite={noop}
              setActiveOffer={noop}
              match={match}
            ><React.Fragment />
            </RoomScreen>
          </BrowserRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
