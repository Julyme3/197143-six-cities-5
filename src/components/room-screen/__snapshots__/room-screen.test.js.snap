// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomScreen render RoomScreen for Authorization user 1`] = `
<MainLayout
  classNameWrap="page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            The house among olive 
          </h1>
          <button
            className="property__bookmark-button button
                      "
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <Rating
            className="property__stars"
            rating={3.8}
          />
          <span
            className="property__rating-value rating__value"
          >
            3.8
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            house
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            3
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            4
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            493
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          />
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Laptop friendly workspace
            </li>
            <li
              className="property__inside-item"
            >
              Washer
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews ·
            <span
              className="reviews__amount"
            >
              2
            </span>
          </h2>
          <ReviewsList
            reviews={
              Array [
                Object {
                  "date": "2020-11-05T13:38:44.753Z",
                  "dateFormated": "November 2020",
                  "id": 3,
                  "name": "Isaac",
                  "raiting": 5,
                  "src": "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg",
                  "text": "The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.",
                },
                Object {
                  "date": "2020-11-05T13:38:44.753Z",
                  "dateFormated": "November 2020",
                  "id": 2,
                  "name": "Christina",
                  "raiting": 4,
                  "src": "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg",
                  "text": "I stayed here for one night and it was an unpleasant experience.",
                },
              ]
            }
          />
          <ReviewForm
            commentValue=""
            isDisabled={true}
            onFieldChange={[Function]}
            onSubmit={[Function]}
            ratingValue=""
          />
        </section>
      </div>
    </div>
    <section
      className="property__map map"
    >
      <Map
        activeCardId={2}
        cityCoords={
          Array [
            50.938361,
            6.959974,
          ]
        }
        height="579px"
        offers={
          Array [
            Object {
              "city": Object {
                "location": Array [
                  50.938361,
                  6.959974,
                ],
                "name": "Cologne",
                "zoom": 13,
              },
              "coords": Array [
                50.957361,
                6.9509739999999995,
              ],
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "details": Object {
                "adults": 4,
                "bedrooms": 3,
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "insideItems": Array [
                  "Laptop friendly workspace",
                  "Washer",
                ],
              },
              "id": 1,
              "isBookmark": false,
              "isPremium": false,
              "name": "The house among olive ",
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "price": 493,
              "raiting": 3.8,
              "src": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              ],
              "type": "house",
              "typeComponent": "near",
              "user": Object {
                "id": 25,
                "isSuper": true,
                "name": "Angelina",
                "src": "img/avatar-angelina.jpg",
              },
            },
            Object {
              "city": Object {
                "location": Array [
                  50.938361,
                  6.959974,
                ],
                "name": "Cologne",
                "zoom": 13,
              },
              "coords": Array [
                50.957361,
                6.9509739999999995,
              ],
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "details": Object {
                "adults": 4,
                "bedrooms": 3,
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "insideItems": Array [
                  "Laptop friendly workspace",
                  "Washer",
                ],
              },
              "id": 2,
              "isBookmark": false,
              "isPremium": false,
              "name": "The house among olive ",
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "price": 493,
              "raiting": 3.8,
              "src": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              ],
              "type": "house",
              "typeComponent": "near",
              "user": Object {
                "id": 25,
                "isSuper": true,
                "name": "Angelina",
                "src": "img/avatar-angelina.jpg",
              },
            },
            Object {
              "city": Object {
                "location": Array [
                  50.938361,
                  6.959974,
                ],
                "name": "Cologne",
                "zoom": 13,
              },
              "coords": Array [
                50.957361,
                6.9509739999999995,
              ],
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "details": Object {
                "adults": 4,
                "bedrooms": 3,
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "insideItems": Array [
                  "Laptop friendly workspace",
                  "Washer",
                ],
              },
              "id": 1,
              "isBookmark": false,
              "isPremium": false,
              "name": "The house among olive ",
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "price": 493,
              "raiting": 3.8,
              "src": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              ],
              "type": "house",
              "typeComponent": "main",
              "user": Object {
                "id": 25,
                "isSuper": true,
                "name": "Angelina",
                "src": "img/avatar-angelina.jpg",
              },
            },
          ]
        }
        width="100%"
        zoom={13}
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <OfferListNear
        activeItem={null}
        offers={
          Array [
            Object {
              "city": Object {
                "location": Array [
                  50.938361,
                  6.959974,
                ],
                "name": "Cologne",
                "zoom": 13,
              },
              "coords": Array [
                50.957361,
                6.9509739999999995,
              ],
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "details": Object {
                "adults": 4,
                "bedrooms": 3,
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "insideItems": Array [
                  "Laptop friendly workspace",
                  "Washer",
                ],
              },
              "id": 1,
              "isBookmark": false,
              "isPremium": false,
              "name": "The house among olive ",
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "price": 493,
              "raiting": 3.8,
              "src": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              ],
              "type": "house",
              "typeComponent": "near",
              "user": Object {
                "id": 25,
                "isSuper": true,
                "name": "Angelina",
                "src": "img/avatar-angelina.jpg",
              },
            },
            Object {
              "city": Object {
                "location": Array [
                  50.938361,
                  6.959974,
                ],
                "name": "Cologne",
                "zoom": 13,
              },
              "coords": Array [
                50.957361,
                6.9509739999999995,
              ],
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "details": Object {
                "adults": 4,
                "bedrooms": 3,
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "insideItems": Array [
                  "Laptop friendly workspace",
                  "Washer",
                ],
              },
              "id": 2,
              "isBookmark": false,
              "isPremium": false,
              "name": "The house among olive ",
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "price": 493,
              "raiting": 3.8,
              "src": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              ],
              "type": "house",
              "typeComponent": "near",
              "user": Object {
                "id": 25,
                "isSuper": true,
                "name": "Angelina",
                "src": "img/avatar-angelina.jpg",
              },
            },
          ]
        }
        onChangeActiveItem={[Function]}
      />
    </section>
  </div>
</MainLayout>
`;

exports[`RoomScreen render RoomScreen for no Authorization user 1`] = `
<MainLayout
  classNameWrap="page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            The house among olive 
          </h1>
          <button
            className="property__bookmark-button button
                      "
            onClick={[Function]}
            type="button"
          >
            <svg
              className="place-card__bookmark-icon property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <Rating
            className="property__stars"
            rating={3.8}
          />
          <span
            className="property__rating-value rating__value"
          >
            3.8
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            house
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            3
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            4
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            493
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          />
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Laptop friendly workspace
            </li>
            <li
              className="property__inside-item"
            >
              Washer
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews ·
            <span
              className="reviews__amount"
            >
              2
            </span>
          </h2>
          <ReviewsList
            reviews={
              Array [
                Object {
                  "date": "2020-11-05T13:38:44.753Z",
                  "dateFormated": "November 2020",
                  "id": 3,
                  "name": "Isaac",
                  "raiting": 5,
                  "src": "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/3.jpg",
                  "text": "The room was spacious and clean. The pool looked nothing like the photos and desparately needs a clean. The sauna and spa were closed for lunar new year holiday.",
                },
                Object {
                  "date": "2020-11-05T13:38:44.753Z",
                  "dateFormated": "November 2020",
                  "id": 2,
                  "name": "Christina",
                  "raiting": 4,
                  "src": "https://assets.htmlacademy.ru/intensives/javascript-3/avatar/10.jpg",
                  "text": "I stayed here for one night and it was an unpleasant experience.",
                },
              ]
            }
          />
        </section>
      </div>
    </div>
    <section
      className="property__map map"
    >
      <Map
        activeCardId={2}
        cityCoords={
          Array [
            50.938361,
            6.959974,
          ]
        }
        height="579px"
        offers={
          Array [
            Object {
              "city": Object {
                "location": Array [
                  50.938361,
                  6.959974,
                ],
                "name": "Cologne",
                "zoom": 13,
              },
              "coords": Array [
                50.957361,
                6.9509739999999995,
              ],
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "details": Object {
                "adults": 4,
                "bedrooms": 3,
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "insideItems": Array [
                  "Laptop friendly workspace",
                  "Washer",
                ],
              },
              "id": 1,
              "isBookmark": false,
              "isPremium": false,
              "name": "The house among olive ",
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "price": 493,
              "raiting": 3.8,
              "src": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              ],
              "type": "house",
              "typeComponent": "near",
              "user": Object {
                "id": 25,
                "isSuper": true,
                "name": "Angelina",
                "src": "img/avatar-angelina.jpg",
              },
            },
            Object {
              "city": Object {
                "location": Array [
                  50.938361,
                  6.959974,
                ],
                "name": "Cologne",
                "zoom": 13,
              },
              "coords": Array [
                50.957361,
                6.9509739999999995,
              ],
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "details": Object {
                "adults": 4,
                "bedrooms": 3,
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "insideItems": Array [
                  "Laptop friendly workspace",
                  "Washer",
                ],
              },
              "id": 2,
              "isBookmark": false,
              "isPremium": false,
              "name": "The house among olive ",
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "price": 493,
              "raiting": 3.8,
              "src": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              ],
              "type": "house",
              "typeComponent": "near",
              "user": Object {
                "id": 25,
                "isSuper": true,
                "name": "Angelina",
                "src": "img/avatar-angelina.jpg",
              },
            },
            Object {
              "city": Object {
                "location": Array [
                  50.938361,
                  6.959974,
                ],
                "name": "Cologne",
                "zoom": 13,
              },
              "coords": Array [
                50.957361,
                6.9509739999999995,
              ],
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "details": Object {
                "adults": 4,
                "bedrooms": 3,
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "insideItems": Array [
                  "Laptop friendly workspace",
                  "Washer",
                ],
              },
              "id": 1,
              "isBookmark": false,
              "isPremium": false,
              "name": "The house among olive ",
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "price": 493,
              "raiting": 3.8,
              "src": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              ],
              "type": "house",
              "typeComponent": "main",
              "user": Object {
                "id": 25,
                "isSuper": true,
                "name": "Angelina",
                "src": "img/avatar-angelina.jpg",
              },
            },
          ]
        }
        width="100%"
        zoom={13}
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <OfferListNear
        activeItem={null}
        offers={
          Array [
            Object {
              "city": Object {
                "location": Array [
                  50.938361,
                  6.959974,
                ],
                "name": "Cologne",
                "zoom": 13,
              },
              "coords": Array [
                50.957361,
                6.9509739999999995,
              ],
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "details": Object {
                "adults": 4,
                "bedrooms": 3,
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "insideItems": Array [
                  "Laptop friendly workspace",
                  "Washer",
                ],
              },
              "id": 1,
              "isBookmark": false,
              "isPremium": false,
              "name": "The house among olive ",
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "price": 493,
              "raiting": 3.8,
              "src": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              ],
              "type": "house",
              "typeComponent": "near",
              "user": Object {
                "id": 25,
                "isSuper": true,
                "name": "Angelina",
                "src": "img/avatar-angelina.jpg",
              },
            },
            Object {
              "city": Object {
                "location": Array [
                  50.938361,
                  6.959974,
                ],
                "name": "Cologne",
                "zoom": 13,
              },
              "coords": Array [
                50.957361,
                6.9509739999999995,
              ],
              "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
              "details": Object {
                "adults": 4,
                "bedrooms": 3,
                "description": "Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.",
                "insideItems": Array [
                  "Laptop friendly workspace",
                  "Washer",
                ],
              },
              "id": 2,
              "isBookmark": false,
              "isPremium": false,
              "name": "The house among olive ",
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg",
              "price": 493,
              "raiting": 3.8,
              "src": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg",
              ],
              "type": "house",
              "typeComponent": "near",
              "user": Object {
                "id": 25,
                "isSuper": true,
                "name": "Angelina",
                "src": "img/avatar-angelina.jpg",
              },
            },
          ]
        }
        onChangeActiveItem={[Function]}
      />
    </section>
  </div>
</MainLayout>
`;
